pandas/plotting/_core.py

```diff
@@ -829,11 +831,30 @@ def _post_plot_logic(self, ax, data):
 class ScatterPlot(PlanePlot):
     _kind = 'scatter'
 
-    def __init__(self, data, x, y, s=None, c=None, **kwargs):
+    def __init__(self, data, x, y, s=None, c=None, size_factor=1, **kwargs):
         if s is None:
-            # hide the matplotlib default for size, in case we want to change
-            # the handling of this argument later
+            # Set default size if no argument is given.
             s = 20
+        elif is_hashable(s) and s in data.columns:
```

Comment:
What does the hashable check do for us here?

Reply:
The hashability check ensures backward compatibility: it is possible (in the current stable version of pandas) to pass an array or a Series of sizes to s. For instance you may do something like:

```python
df.plot.scatter(x='height', y='weight', s=df['price'])
```

Without hashability check, `if s in data.columns` throws an error if s is an array or Series. The hashability check ensures that `if s in data.columns` will only be run if s is a legitimate column label.
If there is a better way to handle this, let me know.

Reply:
So what happens here if a series is passed into the method? I feel like there is something missing from the if...else logic but I could be wrong